### ISSUE TITLE
Detection of loops in table definitions

### ISSUE THREAD

**Issue by sakoay** [datajoint/datajoint-matlab#149]
If a dumb user like myself specifies table A to be a key of table B and also table B to be a key of table A, the DJ parsing seems to go into an infinite loop. Not a very high priority request but would be nice to detect this situation and throw an error instead.

**Comment by dimitri-yatsenko** [comment]
Yes, DataJoint schemas should be acyclic. If it goes into a loop, it should raise an informative error.
